armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -225,8 +225,51 @@ class can then use to modify the isotopics as necessary.
             constructedObject = components.factory(shape, [], kwargs)
             _setComponentFlags(constructedObject, self.flags, blueprint)
             insertDepletableNuclideKeys(constructedObject, blueprint)
+
+        # set the custom density for non-custom material components after construction
+        self.setCustomDensity(constructedObject, blueprint, matMods)
+
         return constructedObject
 
+    def setCustomDensity(self, constructedComponent, blueprint, matMods):
+        """Apply a custom density to a material with custom isotopics but not a 'custom material'."""
+        if self.isotopics is not None:
+            # must nest because one can't check for attribues on a NoneType
+            if blueprint.customIsotopics[self.isotopics].density > 0:
```

Comment:
These two `if` blocks make the entire rest of the method double-indented. Another option would be to do:

```python
        if self.isotopics is None:
            return

    if blueprint.customIsotopics[self.isotopics].density <= 0:
            return
```

This might make the rest of the method easier to read. But I'm not going to die on any hills today, I'm just pointing it out for the future.

Reply:
This is a good suggestion but it makes me realize that a user specifying a zero or negative density is not handled. I will be pushing something up that handles said case.